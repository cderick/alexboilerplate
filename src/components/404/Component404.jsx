import React, { Component } from 'react'
import s from './Component404.scss'
import messages from './data.json'

class Component404 extends Component {
    render() {
        if(!messages) return null
        const { title, subtitle, button } = messages
        return (
            <section className={`container ${s.containerMin} p-0`}>
                <div className="row">
                    <div className="col">
                       {title && <h2 className="pb-4 mb-0 text-muted">{title} <i className="far ml-4 text-danger fa-hand-paper" /></h2>} 
                        {subtitle && <p className="pb-5">{subtitle}</p>}
                        {button && <button onClick={() => window.location.href = '/'} className={s.standardButton}>{button}</button>}
                    </div>
                </div>
            </section>
        )
    }
}

export default Component404