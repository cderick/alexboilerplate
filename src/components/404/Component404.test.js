import React from 'react'
import renderer from 'react-test-renderer'
import Component404 from './Component404'
import { MockedProvider } from '@apollo/react-testing'


describe('Test on <Component404 /> container', () => {
  test('should render with no props', () => {
    const componentNoprops = renderer.create(
      <MockedProvider mocks={[]}>
        <Component404 />
      </MockedProvider>,
    )
    const treeNoprops = componentNoprops.toJSON()
    expect(treeNoprops).toMatchSnapshot()
  })
})

