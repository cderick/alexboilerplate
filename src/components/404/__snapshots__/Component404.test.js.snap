// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <Component404 /> container should render with no props 1`] = `
<section
  className="container containerMin p-0"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h2
        className="pb-4 mb-0 text-muted"
      >
        Sorry, we couldn't find the page you're looking for! 
        <i
          className="far ml-4 text-danger fa-hand-paper"
        />
      </h2>
      <p
        className="pb-5"
      >
        Try searching for new products above, or return to our Homepage.
      </p>
      <button
        className="standardButton"
        onClick={[Function]}
      >
        Homepage
      </button>
    </div>
  </div>
</section>
`;
